### ISSUE TITLE
Getting Property 'x' does not exist on type .... .Vetur(2339)

### ISSUE THREAD

**Issue by jd1378** [vuejs/vetur#1834]
- [X] I have searched through existing issues
- [X] I have read through [docs](https://vuejs.github.io/vetur)
- [X] I have read [FAQ](https://github.com/vuejs/vetur/blob/master/docs/FAQ.md)

## Info

- Platform: Win
- Vetur version: 0.24.0
- VS Code version:1.44.0-insider

## Problem

adding properties inside a component breaks the known data and computed props and says `Property 'x' does not exist on type 'CombinedVueInstance<Vue, object, object, object, Record<never, any>>'.Vetur(2339)`

![break](https://user-images.githubusercontent.com/6196533/78858866-e1073e80-7a42-11ea-97e0-7e9e380ccc2d.gif)

this can be a potential issue on next release of vscode maybe ?

## Reproducible Case
Use veturpack and newest vscode-insiders
I can confirm it works properly with current stable release of vscode

**Comment by jd1378** [comment]
it seems to be fixed now

**Comment by red-meadow** [comment]
Setting "vetur.experimental.templateInterpolationService" to `false` solved this issue for me.

**Comment by octref** [comment]
If you are having computed properties, you would need to annotate them: https://vuejs.org/v2/guide/typescript.html#Annotating-Return-Types

Otherwise TypeScript is unable to infer `this` type.
Even though you are writing JS, Vetur is using TS to support it.

**Comment by Petercopter** [comment]
Weird, this just started happening to me today. MacOS Catalina. Setting `vetur.experimental.templateInterpolationService` like @red-meadow suggested doesn't work for me. It was disabled already, and toggling doesn't seem to have any effect.

I wonder what has changed where to cause this...

**Comment by achimdan** [comment]
me too

**Comment by rchl** [comment]
There is a new setting called `vetur.validation.interpolation` in 0.27.

**Comment by darzove** [comment]
Having this issue as well, `vetur.experimental.templateInterpolationService` was disabled. Code works perfectly fine, but the code minimap gets lit up like a Christmas tree. At least it's in season I guess.

**Comment by mkidsc1603** [comment]
disable vetur.validation.interpolation . Waring is not show, but hint is not work :disappointed:

**Comment by xingorg1** [comment]
Setting "vetur.validation.script" to false solved this issue for me.

**Comment by gabrielxpinheiro** [comment]
Change to false "vetur.validation.script" solved for me. Thanks.

**Comment by ShubhamVerma1811** [comment]
> 
> 
> Setting "vetur.validation.script" to false solved this issue for me.
This seemed to solve to squiggly lines.
But the intellisense is still missing for me until I write `this.$xyz`.

**Comment by rchl** [comment]
You can't disable validation and still have intellisense. Those go together.

Enable validation and fix type issue instead. How? It depends on your code but typically you have to add type annotations in various places.

Also see https://github.com/vuejs/vetur/blob/master/docs/guide/FAQ.md#property-xxx-does-not-exist-on-type-combinedvueinstance

**Comment by ShubhamVerma1811** [comment]
@rchl 
Oh I didn't knew that
But I am facing the error in my script tag
Doing `Object.keys(context)` in `asyncData` shows that I have the `$axios` hook present, but doing `context.$` shows no types

It weird why my types arent working

**Comment by rchl** [comment]
Types is not a runtime thing. What you see at runtime is not gonna match types if you don't have types set up properly.

That's not a problem related to Vetur. Please ask for help in Nuxt discord, for example (https://discord.com/invite/ps2h6QT).

**Comment by vanviethieuanh** [comment]
![image](https://user-images.githubusercontent.com/29355962/116136268-47b5d380-a6fc-11eb-9015-7c4d978e9c76.png)

I restart my computer for n times :<, tried all you guy say and nothing seems work :<

**Comment by topvis** [comment]
![image](https://user-images.githubusercontent.com/5996531/116800756-f3cb3600-aad1-11eb-9743-3669a97d67d6.png)
Unchecking Vetur > Validation: Interpolation resolved the issue for me.

**Comment by microJ** [comment]
in my case, set prop type as `PropType<T>` and return default value as `T` work for me. 😀

```js
props: {
    detail: {
      type: Object as PropType<IDetail>,
      default() {
        return {} as IDetail
      },
    },
  }
```

**Comment by rcristaldi** [comment]
"vetur.experimental.templateInterpolationService": false

**Comment by eaardal** [comment]
Lots of workarounds for turning of the warnings here but I wanted to find a solution for making TypeScript/vetur accept that I had extended the type with my own fields. 

In my example I was installing [nuxt-ssr-screen-size](https://www.npmjs.com/package/nuxt-ssr-screen-size) as a plugin and vetur was complaining that `this.$vssWidth` and `this.$vssHeight` was not a part of CombinedVueInstance and bla bla bla.

Thankfully, Vue's documentation explains how to solve this here: https://vuejs.org/v2/guide/typescript.html#Augmenting-Types-for-Use-with-Plugins

(Although I use Vue 2, here is some docs for Vue 3 with TypeScript: https://v3.vuejs.org/guide/typescript-support.html#typescript-support)

In short, I did this:

1. Created `shims.d.ts` with

```typescript
/* eslint-disable @typescript-eslint/no-unused-vars */

import Vue from 'vue'

declare module 'vue/types/vue' {
  interface Vue {
    $vssWidth: number
    $vssHeight: number
  }
}
```
2. In `tsconfig.json` I added `./src/shims.d.ts` to the `types` array: 
```
"types": ["@nuxt/types", "@types/node", "./src/shims.d.ts"]
```

Now TypeScript and vetur agrees that `this.$vssWidth` and `this.$vssHeight` is part of `this` and there is no need to tweak vetur's options or rules.

**Comment by sarrietav-dev** [comment]
I had the following error threw by Vetur.
![image](https://user-images.githubusercontent.com/25210925/126935570-28c8e79e-5d84-4dab-ba89-6ac5772e8a4d.png)

Following what @octref said:
> If you are having computed properties, you would need to annotate them (with JSDocs): https://vuejs.org/v2/guide/typescript.html#Annotating-Return-Types

I just gave the method a return type and the error went away.
![image](https://user-images.githubusercontent.com/25210925/126935762-115abc95-0b75-420c-9c0c-a13860bf603b.png)

**Comment by agazib** [comment]
Adding `"vetur.validation.script": false` to settings.json in vscode solved for me.

**Comment by yoyo930021** [comment]
I will lock this issue.
Please check https://vuejs.github.io/vetur/guide/FAQ.html#property-xxx-does-not-exist-on-type-combinedvueinstance
Another problem, please open a new issue.
